src/features/snap/components/ReportSnap.tsx

```diff
@@ -1,21 +1,53 @@
+import { t } from '@lingui/macro';
 import { useState, type FunctionComponent } from 'react';
+import { useAccount } from 'wagmi';
 
 import { ReportSnapModal } from './modals/ReportSnapModal';
 import { ReportButton } from '../../../components';
+import { useVerifiableCredential } from '../../../hooks';
+import { useSignErrorHandler } from '../../../hooks/useSignErrorHandler';
+import useToastMsg from '../../../hooks/useToastMsg';
 
 type SnapReportProps = {
+  snapId: string;
   snapName: string;
 };
 
 export const SnapReport: FunctionComponent<SnapReportProps> = ({
+  snapId,
   snapName,
 }) => {
   const [showModal, setShowModal] = useState(false);
+  // TODO  need to changed in future when we store the reported snaps in redux store.
   const [reported, setReported] = useState(false);
 
-  const onSign = async () => {
-    setShowModal(false);
-    setReported(true);
+  const { address } = useAccount();
+  const { showSuccessMsg } = useToastMsg();
+
+  const { signMessage, signError, snapVCBuilder } = useVerifiableCredential();
+
+  useSignErrorHandler(signError);
+
+  // TODO: hardcode options for now, change to dynamic if needed,
+  // TODO may need to consider change options to key value pair to support i18n, key to stored in DB, value is localized string shown in UI.
+  const options = [t`Scam`, t`Vulnerable`];
+
+  const onSign = async (selected: string[]) => {
+    if (address) {
```

Comment:
Handle no address with an error

Reply:
@dawnseeker8 may be we can pass in the address (connected address) from Snap page, as when you not having connected, this components should not be appear, in that case, you dont need to handle a error in this action, like what other report/endorse workflow did

Reply:
First of all, when user account is not connected, this ReportSnap will never been reached since we have protected code in `{Snap.slug}.tsx` file, therefore, we decide to not show error message in here to prevent creating too much translation string and unreachable code.

Reply:
@stanleyyconsensys ok, i will do the change to pass the address.

Reply:
@stanleyyconsensys i remember why i didnt pass the address from snap screen because snap screen didn't have address, the way i control ReportSnap component hide and show is using `isConnected` flag in useAccount, anyway, i will modify that {snap.slug} screen to include address, and then pass in to reportSnap

Reply:
Sorry, `if(address)` is still needed here to prevent `snapVCBuilder.buildDisputedPayload(address, snapID, selected)`  reporting lint error.

so not matter which way, the above if statement can't be skipped.

Reply:
yes, useAccount can return you the address, you dont need to do it twice
 `const { address: connectedAddress, isConnected } = useAccount();`

Reply:
have done the change on both {Snap.slug} file and 'ReportSnap' file and relative unit tests.